### DISCUSSION TITLE
why queryClient.invalidateQueries response type Promise

### DISCUSSION THREAD

**Discussion by kimjh0813** [TanStack/query#5978] [Q&A]
<img width="723" alt="image" src="https://github.com/TanStack/query/assets/105820839/1c1de494-4ab2-4bf6-be5f-0a2336f4b6b9">

const response = await queryClient.invalidateQueries([DIR_KEYS.FILE_COMMENT]);

response = undefind 
why is the response type Promise, is it just for asynchronous processing?

**Comment by TkDodo** [comment]
yes, you can `await` it to make sure the code after it only runs once the refetches that are potentially triggered by invalidation have completed.

**Comment by Lisba** [comment]
@TkDodo It means that is the same using `await` or `return`
`await queryClient.invalidateQueries([DIR_KEYS.FILE_COMMENT]);`
or
`return queryClient.invalidateQueries([DIR_KEYS.FILE_COMMENT]);`,
because both means that your mutation will wait until all the potentially queries triggered by invalidation have completed, right?
